dotnet/sample/AutoGen.BasicSamples/Example05_Dalle_And_GPT4V.cs

```diff
@@ -66,50 +67,39 @@ public static async Task RunAsync()
             File.Delete(imagePath);
         }
 
-        var dalleAgent = new AssistantAgent(
-            name: "dalle",
-            systemMessage: "You are a DALL-E agent that generate image from prompt, when conversation is terminated, return the most recent image url",
-            llmConfig: new ConversableAgentConfig
-            {
-                Temperature = 0,
-                ConfigList = gpt35Config,
-                FunctionContracts = new[]
-                {
-                    instance.GenerateImageFunctionContract,
-                },
-            },
+        var generateImageFunctionMiddleware = new FunctionCallMiddleware(
+            functions: [instance.GenerateImageFunctionContract],
             functionMap: new Dictionary<string, Func<string, Task<string>>>
             {
                 { nameof(GenerateImage), instance.GenerateImageWrapper },
-            })
+            });
+        var dalleAgent = new OpenAIChatAgent(
+            openAIClient: openAIClient,
+            modelName: "gpt-3.5-turbo",
+            name: "dalle",
+            systemMessage: "You are a DALL-E agent that generate image from prompt, when conversation is terminated, return the most recent image url")
+            .RegisterMessageConnector()
+            .RegisterStreamingMiddleware(generateImageFunctionMiddleware)
             .RegisterMiddleware(async (msgs, option, agent, ct) =>
             {
-                // if last message contains [TERMINATE], then find the last image url and terminate the conversation
-                if (msgs.Last().GetContent()?.Contains("TERMINATE") is true)
+                if (msgs.Any(msg => msg.GetContent()?.ToLower().Contains("approve") is true))
                 {
-                    var lastMessageWithImage = msgs.Last(msg => msg is ImageMessage) as ImageMessage;
-                    var lastImageUrl = lastMessageWithImage.Url;
-                    Console.WriteLine($"download image from {lastImageUrl} to {imagePath}");
-                    var httpClient = new HttpClient();
-                    var imageBytes = await httpClient.GetByteArrayAsync(lastImageUrl);
-                    File.WriteAllBytes(imagePath, imageBytes);
-
-                    var messageContent = $@"{GroupChatExtension.TERMINATE}
-
-{lastImageUrl}";
-                    return new TextMessage(Role.Assistant, messageContent)
-                    {
-                        From = "dalle",
-                    };
+                    return new TextMessage(Role.Assistant, $"The image satisfies the condition, conversation is terminated. {GroupChatExtension.TERMINATE}");
                 }
 
-                var reply = await agent.GenerateReplyAsync(msgs, option, ct);
+                var msgsWithoutImage = msgs.Where(msg => msg is not ImageMessage).ToList();
+                var reply = await agent.GenerateReplyAsync(msgsWithoutImage, option, ct);
```

Comment:
If `reply.GetContent()` can return null, would it crash ?

Reply:
In this case yes, because an empty content basically means something wrong in the llm call.